src/AzModuleInstaller.ts

```diff
@@ -0,0 +1,159 @@
+        this.version = version;
+        this.githubAuth = githubAuth;
+        this.installResult = {
+            moduleSource: "Others",
+            isInstalled: false
+        };
+        const platform = (process.env.RUNNER_OS || os.type())?.toLowerCase();
+        switch(platform) {
+            case "windows":
+            case "windows_nt":
+                this.isWin = true;
+                this.moduleContainerPath = "C:\\Modules";
+                this.modulePath = `${this.moduleContainerPath}\\az_${this.version}`
+                break;
+            case "linux":
+                this.moduleContainerPath = "/usr/share";
+                this.modulePath = `${this.moduleContainerPath}/az_${this.version}`
+                break;
+            default:
+                throw `OS ${platform} not supported`;
+        }
+        this.moduleZipPath = `${this.modulePath}.zip`;
+    }
+
+    public async install(): Promise<InstallResult> {
+        if (Utils.isHostedAgent(this.moduleContainerPath)) {
+            await this.tryInstallingLatest();
+            await this.tryInstallFromFolder();
+            await this.tryInstallFromZip();
+            await this.tryInstallFromGHRelease();
+            await this.tryInstallFromPSGallery();
+        } else {
+            core.debug("File layout is not like hosted agent, skippig module install.");
+            this.installResult = {
+                isInstalled: false,
+                moduleSource: "privateAgent"
+            };
+        }
+
+        return this.installResult;
+    }
+
+    private async tryInstallingLatest() {
+        if (this.installResult.isInstalled) {
+            return;
+        }
+
+        if (this.version === "latest") {
+            core.debug("Latest selected, will use latest Az module available in agent as folder.");
+            this.installResult = {
+                isInstalled: true,
+                moduleSource: "hostedAgentFolder"
+            };
+        }
+    }
+
+    private async tryInstallFromFolder() {
+        if (this.installResult.isInstalled) {
+            return;
+        }
+
+        if (FileUtils.pathExists(this.modulePath)) {
+            core.debug(`Az ${this.version} present at ${this.modulePath} as folder.`);
+            this.installResult = {
+                isInstalled: true,
+                moduleSource: "hostedAgentFolder"
+            };
+        }
+    }
+
+    private async tryInstallFromZip() {
+        if (this.installResult.isInstalled) {
+            return;
+        }
+
+        if (FileUtils.pathExists(this.moduleZipPath)) {
+            core.debug(`Az ${this.version} present at ${this.moduleZipPath} as zip, expanding it.`);
+            await new ArchiveTools(this.isWin).unzip(this.moduleZipPath, this.moduleContainerPath);
+            await FileUtils.deleteFile(this.moduleZipPath);
+            this.installResult = {
+                isInstalled: true,
+                moduleSource: "hostedAgentZip"
```

Comment:
Can you put these values of moduleSource in a Constants file

Reply:
done